### PR TITLE
feat(ui-shell): add theming support 

### LINKED ISSUE #8102: [Design] UI Shell - new color token spec
Spec for the UI Shell using the [new color tokens](https://www.notion.so/carbondesignsystem/Final-List-of-new-tokens-fc5985c7331341baa6546064399d36da). 

UI Shell – Header
UI Shell – Side-nav

### LINKED ISSUE #8409: Theme the UI Shell
Epic to track the UI Shell Theming effort

Design specs: https://github.com/carbon-design-system/carbon/issues/8102

We will need to update all UI Shell scss files to use the new color tokens, and remove the old tokens. We should also see if there is any unused code that is no longer needed

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #8409 
Closes #8102 

This is the first step in creating our UI Shell Theme 

- [x] Update `_header.scss` files to match most recent styles
- [x] Update `_side-nav.scss` files to match most recent styles
- [x] Update `_switcher.scss` files to match most recent styles
- [x] Update `_header-panel.scss` files to match most recent styles
- [x] Update `_navigation-menu.scss` files to match most recent styles
- [x] Update `_product-switcher.scss` files to match most recent styles

Deprecated the following components that did not have a design spec (per discussion with design):
- [x] HeaderPanel + accompanying styles
- [x] SideNavDetails + accompanying styles
- [x] SideNavFooter + accompanying styles
- [x] SideNavHeader + accompanying styles
- [x] SideNavSwitcher + accompanying styles
- [x] navigation-menu styles (which did not have a component built out)
- [x] custom UI-shell theme tokens

### CODE REVIEW

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 1]
```diff
@@ -34,6 +37,15 @@ const HeaderPanel = React.forwardRef(function HeaderPanel(
     [customClassName]: !!customClassName,
   });
 
+  if (__DEV__) {
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/HeaderPanel.js]: I think in the recent refactor of `warning()` they included this condition on `__DEV__` internally so we can unconditionally call `warning()` in the components.

**Review by aagonzales — COMMENTED** [review]
Its looking good! Just a few changes.

The selected state for header actions isn't showing when the panel is open
image

The menu items in the header should be using `$layer` for its background color.
image

The icon in a selected item in the left nav should be using `$icon-01`/`$icon-primary`
image

**Comment by andreancardona** [comment]
@aagonzales @tay1orjones @tw15egan finally ready for another review :)

**Review by aagonzales — APPROVED** [review]
Looks good to me now! 🎉

**Comment by tw15egan** [comment]
This looks awesome! I think the only issue I'm seeing is that the `bx--content` class doesn't set text color, so the text gets washed out in dark themes:

Screen Shot 2021-05-20 at 12 26 54 PM


I was able to fix this by adding  `color: $text-primary;` on `L19` in `_content.scss`

**Comment by andreancardona** [comment]
@tw15egan thanks for catching that - just updated :)

**Comment by tw15egan** [comment]
Do we need to hold off on merging this until we have a v11 branch? Seems like this could cause unexpected changes to teams using the UI Shell in g10 / white themes and suddenly having a light header. I believe they would need to be using custom properties to revert back to the dark header, which could be a breaking change to enable for some teams

**Comment by andreancardona** [comment]
> d off on merging this until we have a v11 branch? Seems like this could cause unexpected changes to teams using the UI Shell in g10 / white themes and suddenly having a light header. I believe they

Yep I agree we should hold off..

**Comment by jnm2377** [comment]
Should we change the base to `next` now that we have a next branch?

**Comment by tw15egan** [comment]
@jnm2377 yeah let's go for it 🎸

**Review by tw15egan — APPROVED** [review]

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 2]
```diff
@@ -34,6 +37,15 @@ const HeaderPanel = React.forwardRef(function HeaderPanel(
...
+    warning(
+      didWarnAboutDeprecation,
+      'The `HeaderPanel` component has been deprecated and will be removed ' +
+        'in the next major release of `carbon-components-react`'
+    );
+    didWarnAboutDeprecation = true;
+  }
+
```
Comment (tay1orjones) [inline packages/react/src/components/UIShell/HeaderPanel.js]: Missed just this one `__DEV__`, the others look great 👍 

```suggestion
  warning(
    didWarnAboutDeprecation,
    'The `HeaderPanel` component has been deprecated and will be removed ' +
      'in the next major release of `carbon-components-react`'
  );
  didWarnAboutDeprecation = true;

```

**Review by tay1orjones — APPROVED** [review]
One change I suggested but looks good otherwise!

**Comment by andreancardona** [comment]
> One change I suggested but looks good otherwise!

thanks for catching that!

**Review by jnm2377 — APPROVED** [review]

**Comment by jnm2377** [comment]
Just for tracking purposes, this was never merged into main. After this PR was included in `/next` we started working on the new styles package for v11, and the new tokens were then added to UI shell stylesheets there in this PR https://github.com/carbon-design-system/carbon/pull/9195

**Pull request merged**